### PR TITLE
Alerting: Fix calculation of the matching routes for notification policies when trying to match missing label

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fixes the calculation of the matching routes when notification policy matches missing label.

This error is shown in:
- Notification policy view
- when previewing routes in alert form

as both cases use the same algorithm.

**Who is this feature for?**

All users

**Which issue(s) does this PR fix?**:

Fixes (https://github.com/grafana/grafana/issues/75196)

**Special notes for your reviewer:**

https://github.com/grafana/grafana/assets/33540275/0bcbbbb3-3aab-4d55-9d50-c4a6ffbe76b0

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/utils/notification-policies.ts [hunk 1]
```diff
@@ -48,20 +48,34 @@ interface MatchingResult {
   labelsMatch: Map<Label, LabelMatchResult>;
 }
 
+function isMissingLabelMatcher(matcher: ObjectMatcher): boolean {
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/utils/notification-policies.ts:51]: not sure if there is any other case for matching missing labels?

File: public/app/features/alerting/unified/utils/notification-policies.ts [hunk 2]
```diff
@@ -70,12 +84,30 @@ function matchLabels(matchers: ObjectMatcher[], labels: Label[]): MatchingResult
...
+        if (labelsInInstance.some((label) => isNotEmptyValueInMissingLabel(matcher[0], label))) {
+          return false;
+        }
+        labelMatch = { match: true, matchers: [matcher] };
+        labelMatch.matchers.push(matcher);
+        matchesEmpty = true;
+        labelsMatch.set([matcher[0], ''], labelMatch);
+        return true;
+      } else {
+        return false;
```
Comment (VikaCep) [inline public/app/features/alerting/unified/utils/notification-policies.ts:105]: I'do an early return instead of this `else`

**Review by VikaCep — COMMENTED** [review]
The solution works and LGTM but I'll defer the approval to @gillesdemey as he's worked more on this code

**Comment by gillesdemey** [comment]
Closing this one in favor of https://github.com/grafana/grafana/pull/77292 which should work for any negative label matcher

**Pull request closed without merging**